data-pipeline/user-cache-updater/src/main/java/org/ekstep/ep/samza/service/UserCacheUpdaterService.java

```diff
@@ -0,0 +1,181 @@
+                        Map<String, Object> userLocationMap = getLocationInfo(locationDetails);
+
+                        if (!userLocationMap.isEmpty())
+                            userCacheData.putAll(userLocationMap);
+                    }
+                }
+            }
+            else if("Create".equals(userState)) {
+                userCacheData.putAll(getUserSigninType(event));
+            }
+
+            if (!userCacheData.isEmpty()) {
+                redisConnect.addJsonToCache(userId, gson.toJson(userCacheData), userDataStoreConnection);
+                sink.success();
+            }
+
+        } catch (JedisException ex) {
+            sink.error();
+            LOGGER.error("", "Exception when adding to user redis cache", ex);
+            redisConnect.resetConnection();
+            try (Jedis redisConn = redisConnect.getConnection(userStoreDb)) {
+                this.userDataStoreConnection = redisConn;
+                if (null != userCacheData)
+                    redisConnect.addJsonToCache(userId, gson.toJson(userCacheData), userDataStoreConnection);
+            }
+        }
+        return userCacheData;
+    }
+
+    private Map<String, String> getUserSigninType(Event event) {
+        Map<String,String> userData = new HashMap<>();
+        String signIn_type = null != event.getUserSignInType() ? event.getUserSignInType() : userCacheUpdaterConfig.getUserSignInTypeDefault();
+        if (!(userCacheUpdaterConfig.getUserSignInTypeDefault().equalsIgnoreCase(signIn_type))) {
+            if (userCacheUpdaterConfig.getUserSelfSignedInTypeList().contains(signIn_type)) {
+                userData.put("usersignintype", userCacheUpdaterConfig.getUserSelfSignedKey());
+            } else if (userCacheUpdaterConfig.getUserValidatedTypeList().contains(signIn_type)) {
+                userData.put("usersignintype", userCacheUpdaterConfig.getUserValidatedKey());
+            }
+        }
+        return userData;
+    }
+
+    private Map<String, String> getUserLogininType(Event event) {
+        Map<String,String> userData = new HashMap<>();
+        String loginIn_type = null != event.getUserLoginType() ? event.getUserLoginType() : userCacheUpdaterConfig.getUserLoginInTypeDefault();
+        if (!(userCacheUpdaterConfig.getUserLoginInTypeDefault().equalsIgnoreCase(loginIn_type)))
+            userData.put("userlogintype", loginIn_type);
+        return userData;
+    }
+
+    private <T> List<Row> fetchFallbackDetailsFromDB(T id, String table){
+        String metadataQuery = QueryBuilder.select().all()
+                .from(userCacheUpdaterConfig.cassandra_db(), table)
+                .where(QueryBuilder.eq("id", id))
+                .toString();
+        List<Row> rowSet= null;
+        try {
+            rowSet = cassandraConnection.find(metadataQuery);
+        } catch (DriverException ex) {
```

Comment:
We should have a retry logic here and stop the job if there is a cassandra exception. The job cannot skip the cache updater event